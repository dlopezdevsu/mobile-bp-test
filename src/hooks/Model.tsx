class model_class {

  dispatch: any = null
  get_dispatch = () => this.dispatch == null ? false : this.dispatch
  set_dispatch = (n_dispatch: any) => { this.dispatch = n_dispatch }

  setData = (label: string, payload: any) => {
    this?.dispatch({ type: 'SET_STORE', label, payload })
  }

  resetData = () => {
    this?.dispatch({ type: 'RESET_DATA' })
  }
}

export default new model_class();