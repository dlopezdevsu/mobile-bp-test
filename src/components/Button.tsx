import * as React from 'react';
import { Pressable, PressableProps, StyleSheet, StyleProp, Text } from 'react-native';

import * as Colors from '../utils/colors';

interface ButtonProps extends PressableProps {
  style: StyleProp<any>;
  label: string;
  variant?: 'delete' | 'submit' | 'default';
}

const Button = ({ ...props }: ButtonProps) => {

  return (
    <Pressable
      {...props}
      style={({ pressed }) => [styles.container, styles[props?.variant || 'default'], { opacity: pressed ? 0.5 : 1 }, props.style]}
    >
      <Text style={[styles.label, props?.variant === 'delete' ? styles.textWhite : {}]}>
        {props.label}
      </Text>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.gray,
    padding: 18,
  },
  label: {
    textAlign: 'center',
    fontWeight: '600'
  },
  delete: {
    backgroundColor: Colors.red
  },
  submit: {
    backgroundColor: Colors.yellow
  },
  default: {
    backgroundColor: Colors.gray
  },
  textWhite: {
    color: Colors.white
  }
});

export default Button;