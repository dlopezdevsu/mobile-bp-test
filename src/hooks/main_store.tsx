import React, { ReactElement } from "react";

import { ItemProductListProps } from 'src/components/ItemProductList'

export const Store = React.createContext<any>(null);

export interface InitialStateProps {
  loading: boolean;
  alert: string;
  products: ItemProductListProps[];
  selected: ItemProductListProps | null;
  editMode: boolean;
  openBottomSheet: boolean;
  callbackBottomSheet: () => void;
}

interface ActionProps {
  label: string;
  type: string;
  payload: any;
}

const initialState = {
  loading: true,
  alert: '',
  products: [],
  selected: null,
  editMode: false,
  openBottomSheet: false,
  callbackBottomSheet: () => null,
};

function reducer(state: InitialStateProps, action: ActionProps) {
  switch (action.type) {
    case "SET_STORE":
      return { ...state, [action.label]: action.payload };
    case "RESET_DATA":
      return { ...initialState };
    default:
      return state;
  }
}

export function StoreProvider({ children }: { children: ReactElement }) {
  const [state, dispatch] = React.useReducer(reducer, initialState);
  const value = { state, dispatch };

  return <Store.Provider value={value}>{children}</Store.Provider>;
}
