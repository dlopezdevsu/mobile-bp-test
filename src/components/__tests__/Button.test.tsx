/**
 * @format
 */

import 'react-native';
import React from "react";
import Button from "src/components/Button";
import renderer from "react-test-renderer";
import { StoreProvider } from 'src/hooks/main_store'

import { it, expect } from '@jest/globals';

it(`render Button`, () => {
  const tree = renderer.create(<StoreProvider>
    <Button label='Test button' style={{}} />
  </StoreProvider>);
  expect(tree).toMatchSnapshot();
});
