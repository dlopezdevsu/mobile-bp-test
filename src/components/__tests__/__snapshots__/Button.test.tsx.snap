// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  label="Test button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#eaecf3",
        "padding": 18,
      },
      {
        "backgroundColor": "#eaecf3",
      },
      {
        "opacity": 1,
      },
      {},
    ]
  }
>
  <Text
    style={
      [
        {
          "fontWeight": "600",
          "textAlign": "center",
        },
        {},
      ]
    }
  >
    Test button
  </Text>
</View>
`;
